lib/classes/savings_account.rb

```diff
@@ -0,0 +1,28 @@
+require 'awesome_print'
+module BankAccount
+  class Savings < Account
+
+    def initialize(id, initial_balance)
+      super
+      if @current_balance < 10
+        raise ArgumentError.new, "Initial deposit must be greater than 10"
+      end
+    end
+
+    def withdraw(amount)
+      super
+      if amount > (@current_balance - 10)
```

Comment:
I think you will have a similar issue to what I pointed out in the base Account class because of the fact that you set the `@current_balance` before you check that it will not take the balance too low.